### PR TITLE
Thread `&'b mut F` instead of `F` everywhere

### PR DESCRIPTION
**Pull request by alexcrichton**
I ran into a recursion limit bug when using this in Cargo because
there's a point where `Foo<F>` got turned into `Foo<&mut F>`. Now
everything would compile just fine as `&mut F` still implements `FnMut`,
but in the interest of avoiding `#![recursion_limit]` I figured I'd go
ahead and avoid the recursion in type parameters by just always using
`&mut F` everywhere.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1082,7 +1082,7 @@ impl<'a, X, F> de::MapVisitor for MapVisitor<'a, X, F>
             parent: self.path,
             key: self.key()?,
         };
-        self.delegate.visit_value_seed(TrackedSeed::new(seed, &mut self.callback, path))
+        self.delegate.visit_value_seed(TrackedSeed::new(seed, &mut *self.callback, path))
```
Comment (alexcrichton) [inline src/lib.rs:1085]: This specifically was the line that crucially needed changing (I found at least).

I tried to keep `&'b mut F` out of `Deserializer` itself but it ended up needing to get pushed that high up (I think)

**Comment by dtolnay** [comment]
I added 31ca0e70d7ccf500d1f923c7aebfa2f7a65e02da (that doesn't cancel out your change does it?) and released 0.0.2.

**Comment by alexcrichton** [comment]
Ah yeah I think that works!

Still working through some more issues though, so there may be an 0.0.3 as well :)

**Comment by alexcrichton** [comment]
False alarm, that bug was a missing `#[serde(rename = "...")]`, so 0.0.2 looks like a likely "cargo's gonna use this" candidate!

**Pull request merged**